monte_carlo.py

```diff
@@ -7,43 +7,108 @@
 
 
 class Policy:
+    """
+    A wrapper class for the policy dictonary of the `MonteCarlo` agent.
+    """
+
     def __init__(self, train_epsilon, test_epsilon):
+        """
+        Initializes an instance of the MonteCarlo Reinforcement Learning agent.
+
+        Parameters
+        ----------
+        train_epsilon : float
+            The exploration rate during training (epsilon-greedy). Higher values encourage more exploration.
+        test_epsilon : float
+            The exploration rate during testing (epsilon-greedy). Higher values encourage more exploration.
+        """
+
         self.policy = {}
     
         self.train_epsilon = train_epsilon
         self.test_epsilon = test_epsilon
         self.epsilon = train_epsilon
 
     def __getitem__(self, state):
+        """
+        Returns the actions for the specified `state`.
+
+        Parameters
+        ----------
+        state: tuple(int, int, int, int)
+            The state for which actions should be returned.
+        """
+
         if state not in self.policy:
-            self.policy[state] = np.random.choice(MonteCarlo.get_legal_actions(state))
+            self.policy[state] = np.random.choice(MonteCarlo.get_valid_actions(state))
         if np.random.random() < self.epsilon:
-            return np.random.choice(MonteCarlo.get_legal_actions(state))
+            return np.random.choice(MonteCarlo.get_valid_actions(state))
         return self.policy[state]
     
-    def __setitem__(self, state, action):
-        self.policy[state] = action
+    def __setitem__(self, state, actions):
+        """
+        Sets the `actions` for the specified `state`.
+
+        Parameters
+        ----------
+        state: tuple(int, int, int, int)
+            The state for which `actions` should be set.
+        actions: [str]
+            The actions which should be associated to the specified `state`.
+        """
+
+        self.policy[state] = actions
```

Comment:
```suggestion
    def __setitem__(self, state, action):
        """
        Sets the `action` for the specified `state`.

        Parameters
        ----------
        state: tuple(int, int, int, int)
            The state for which `action` should be set.
        action: str
            The action which should be associated to the specified `state`.
        """

        self.policy[state] = action
```

Why would it be several actions?